bin/workflows/pycbc_make_coinc_search_workflow

```diff
@@ -155,7 +155,23 @@ precalc_psd_files = wf.setup_psd_workflow(workflow, analyzable_segs,
 bank_files = wf.setup_tmpltbank_workflow(workflow, analyzable_segs,
                                          datafind_files, output_dir="bank",
                                          psd_files=precalc_psd_files)
-hdfbank = wf.convert_bank_to_hdf(workflow, bank_files, "bank")
+
+# Determine the extension of the bank file taken as input
+cp = workflow.cp
+tmpltbankMethod = cp.get_opt_tags("workflow-tmpltbank", "tmpltbank-method", '')
+if tmpltbankMethod == "PREGENERATED_BANK":
+    pre_gen_bank = cp.get_opt_tags('workflow-tmpltbank',
+                                           'tmpltbank-pregenerated-bank', '')
+    bank_filename, ext = (pre_gen_bank.split('/')[-1].split('.'))
+
+# If the bank is a hdf or h5 file do not change the format. If not,
+# convert it to hdf
+if ext == 'hdf' or ext == 'h5' :
+    hdfbank = bank_files
+else :
+    hdfbank = wf.convert_bank_to_hdf(workflow, bank_files, "bank")
+logging.info("Input bank is a %s file", ext)
```

Comment:
@soumide1102 Yeah, let me give the motivation here, but mostly what you say is true:

I would like to have all the logic (that you describe) within the `setup_tmpltbank_workflow` function. Other workflows also use that function so having this code in one place avoids potential duplication of code.

*However* PyGRB workflow also uses this function, and that is still using XML template banks, so I would like to have the ability to choose whether the bank is returned in XML or HDF format (or potentially other formats in the future). (If the input bank is in HDF format and the user asks for XML, or if a third format is specified anywhere, just raise a NotImplementedError ... If someone wants to use that code path, they can implement it!)